SOOUM/SOOUM/Presentations/Report/ReportViewController.swift

```diff
@@ -0,0 +1,145 @@
+        
+    let tableViewTitleLabel = UILabel().then {
+        $0.typography = .init(
+            fontContainer: BuiltInFont(
+                size: 16,
+                weight: .semibold
+            ),
+            lineHeight: 22.4,
+            letterSpacing: 0
+        )
+        $0.textColor = .som.black
+        $0.text = "신고 사유 선택"
+    }
+        
+    lazy var reportTableView = UITableView().then {
+        $0.backgroundColor = .clear
+        $0.indicatorStyle = .black
+        $0.separatorStyle = .none
+        
+        $0.register(ReportTableViewCell.self, forCellReuseIdentifier: String(describing: ReportTableViewCell.self))
+        
+        $0.dataSource = self
+        $0.delegate = self
+    }
+    
+    let uploadReportButtonLabel = UILabel().then {
+        $0.typography = .init(
+            fontContainer: BuiltInFont(
+                size: 16,
+                weight: .bold
+            ),
+            lineHeight: 19.1,
+            letterSpacing: 0
+        )
+        $0.text = "신고하기"
+        $0.textColor = .som.white
+        $0.isUserInteractionEnabled = false
+        $0.textAlignment = .center
+        $0.backgroundColor = .som.primary
+        $0.layer.cornerRadius = 10
+        $0.clipsToBounds = true
+    }
+
+    override func setupConstraints() {
+        self.view.addSubview(tableViewTitleLabel)
+        tableViewTitleLabel.snp.makeConstraints {
+            $0.top.equalTo(self.view.safeAreaLayoutGuide.snp.top).offset(28)
+            $0.leading.equalToSuperview().offset(20)
+            $0.height.equalTo(22)
+        }
+        
+        self.view.addSubview(reportTableView)
+        reportTableView.snp.makeConstraints {
+            $0.top.equalTo(tableViewTitleLabel.snp.bottom).offset(22)
+            $0.leading.trailing.equalToSuperview()
+        }
+        
+        self.view.addSubview(uploadReportButtonLabel)
+        uploadReportButtonLabel.snp.makeConstraints {
+            $0.top.equalTo(reportTableView.snp.bottom)
+            $0.bottom.equalTo(self.view.safeAreaLayoutGuide.snp.bottom).offset(-20)
+            $0.leading.equalToSuperview().offset(20)
+            $0.trailing.equalToSuperview().offset(-20)
+            $0.height.equalTo(50)
+        }
+    }
+    
+    override func setupNaviBar() {
+        super.setupNaviBar()
+        self.navigationBar.titleLabel.text = "신고하기"
+        self.navigationBar.isHideBackButton = false
+    }
+    
+    func bind(reactor: ReportViewReactor) {
+        bindAction(reactor: reactor)
+        bindState(reactor: reactor)
+    }
+    
+    private func bindAction(reactor: ReportViewReactor) {
+        uploadReportButtonLabel.rx
+            .tapGesture()
+            .when(.recognized)
+            .compactMap { _ in
+                guard let selectedReason = self.selectedReason else {
+                    return nil
+                }
+                return Reactor.Action.report(selectedReason)
+            }
+            .bind(to: reactor.action)
```

Comment:
```suggestion
    private func bindAction(reactor: ReportViewReactor) {
        uploadReportButtonLabel.rx
            .tapGesture()
            .when(.recognized)
            .compactMap { _ in self.selectedReason }
            .map(Reactor.Action.report)
            .bind(to: reactor.action)
```

위 코드의 compactMap은 옵저버블이 nil이 아닐 때 방출되는 오퍼레이터이기 때문에 위의 상황은 map으로 충분히 대체 가능합니다.
정덕님 의도가 self.selectedReason을 옵셔널 바인딩하고 액션을 방출하는 거라고 이해해서 비슷하게 수정해봤습니다.